core/src/main/scala/com/raphtory/internals/time/DateTimeParser.scala

```diff
@@ -15,11 +19,14 @@ private[raphtory] class DateTimeParser(format: String) {
       .orElse(Try(LocalDate.parse(datetime, formatter).atTime(LocalTime.of(0, 0))))
       .map(_.toInstant(ZoneOffset.UTC).toEpochMilli)
       .get
+
+  def parse(epoch: Long): String = {
+    val dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+    dateFormat.format(new java.util.Date(epoch))
+  }
+
```

Comment:
Fix epoch handling on graph basis (e.g., what about data sets with epoch time and seconds granularity?)